src/Sulu/Component/Cache/MemoizeTwigExtensionTrait.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Cache;
+
+/**
+ * Provides functionality to convert twig functions.
+ */
+trait MemoizeTwigExtensionTrait
+{
+    /**
+     * Convert simple twig functions to use a new context.
+     *
+     * @param \Twig_SimpleFunction[] $functions
+     * @param mixed $context
+     *
+     * @return \Twig_SimpleFunction[]
+     */
+    protected function convertTwigFunctions(array $functions, $context)
```

Comment:
Would actually mean that we need to have knowledge about the `extension` variable in here, but this trait only makes sense if we have wrapped another extension.